### PR TITLE
drivers: display: st7789v: add power management support

### PR DESCRIPTION
**Pull request by endian-albin**
This adds an "enter sleep" function and a power control mechanism to the st7789v display driver. It's been tested on a PineTime DevKit which contains this display bundled with a touch screen.

There already exists a "blanking on" function that gives some power savings (9.7 %) compared to "blanking off" (i.e. "display on"), but sleep mode gives a 686 % improvement (measurements by Lukas Berglund).

Right now, I initialize the device in an active state. I can change that to low power state/sleep mode if you prefer.

The pm_control function largely follows the logic in the power control sample and will set the device in sleep mode if any other mode than DEVICE_PM_ACTIVE_STATE is set. Please let me know if you're unhappy with that behaviour.

### CODE REVIEW

File: drivers/display/display_st7789v.c [hunk 1]
```diff
@@ -88,6 +92,14 @@ static void st7789v_transmit(struct st7789v_data *data, u8_t cmd,
 	}
 }
 
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+static void st7789v_enter_sleep(struct st7789v_data *data)
```
Comment (jfischer-phytec-iot) [inline drivers/display/display_st7789v.c:96]: Can you move `st7789v_enter_sleep()` just above `st7789v_pm_control() ` and so reduce the number of "`#ifdef CONFIG_DEVICE_POWER_MANAGEMENT`", please ?

File: drivers/display/display_st7789v.c [hunk 2]
```diff
@@ -88,6 +92,14 @@ static void st7789v_transmit(struct st7789v_data *data, u8_t cmd,
...
+{
+	st7789v_transmit(data, ST7789V_CMD_SLEEP_IN, NULL, 0);
+	k_sleep(K_MSEC(5));
```
Comment (jfischer-phytec-iot) [inline drivers/display/display_st7789v.c:99]: Why the driver have to wait here for 5 ms?

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: drivers/display/display_st7789v.c (hunk 2 above)
Comment (endian-albin) [reply drivers/display/display_st7789v.c:99]: Thanks! I had another look at the data sheet and the wait should not be necessary. I'll remove it.

File: drivers/display/display_st7789v.c (hunk 1 above)
Comment (endian-albin) [reply drivers/display/display_st7789v.c]: Yes, fixed.

**Review by jfischer-no — APPROVED** [review]

**Review by vanwinkeljan — APPROVED** [review]

**Pull request merged**